sklearn/model_selection/_search.py

```diff
@@ -781,11 +787,16 @@ class GridSearchCV(BaseSearchCV):
             - A string, giving an expression as a function of n_jobs,
               as in '2*n_jobs'
 
-    iid : boolean, default=True
+    iid : boolean, default=None
         If True, the data is assumed to be identically distributed across
         the folds, and the loss minimized is the total loss per sample,
         and not the mean loss across the folds.
 
+        ..deprecated:: 0.19
+            Parameter ``iid`` has been deprecated in version 0.19 and
+            will be removed in 0.21.
+            Future (and default) behavior is equivalent to `iid=true`.
```

Comment:
But isn't @GaelVaroquaux saying that this behaviour is inappropriate? https://github.com/scikit-learn/scikit-learn/pull/9103#issuecomment-307583916

Rather we should be both deprecating the parameter and changing the default.